build/util.rs

```diff
@@ -9,3 +12,18 @@ pub fn to_module_name<P: Into<PathBuf>>(file_name: P) -> String {
         .to_lowercase() // all lowercase
         .replace(|c: char| !c.is_alphanumeric(), "_") // remove non alphanum
 }
+
+/// Removes the trailing zeroes in the payload
+///
+/// # Note:
+///
+/// There must always be at least one remaining byte even if it is a
+/// zero byte.
+pub fn q_remove_trailing_zeroes(buf: proc_macro2::Ident) -> TokenStream {
+    quote! {
+        while let Some(&0) = #buf.last() {
```

Comment:
Doing so we end up duplicating the same code again and again inside the binary, it's better to have a global function that abstract such behavior over having the same logic around every serialized message.

Reply:
To do that, we probably need to move this function to a library that will be used by all dialects.